### ISSUE TITLE
ClassCastException with StringCodec and RSet iterator

### ISSUE THREAD

**Issue by domkun** [mrniko/redisson#168]
I've got a problem with iterating over a RSet, the following code throws a ClassCastException when using the StringCodec:

```java
RSet<Object> test = r.getSet("test");

test.add("test1");
test.add("test2");
test.add("test3");

test.forEach(System.out::println);
```

Stacktrace:

```
java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Number
	at com.lambdaworks.redis.output.ValueSetScanOutput.set(ValueSetScanOutput.java:17)
	at com.lambdaworks.redis.protocol.RedisStateMachine.decode(RedisStateMachine.java:112)
	at com.lambdaworks.redis.protocol.CommandHandler.decode(CommandHandler.java:71)
	at com.lambdaworks.redis.protocol.CommandHandler.channelRead(CommandHandler.java:52)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:339)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:324)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:339)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:324)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:847)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:111)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:137)
	at java.lang.Thread.run(Thread.java:745)
```

With a List everything works.

**Comment by Gladmir** [comment]
Does below code (iterating with for-each instead of java 8 forEach) also throws ClassCastException ?

```
        RSet<Object> someSet = redisson.getSet ("someSet");

        someSet.add ("test1");
        someSet.add ("test2");
        someSet.add ("test3");

        for (Object object : someSet) {
            System.out.println (object);
        }
```

**Comment by mrniko** [comment]
Fixed. Please check

**Comment by Gladmir** [comment]
@mrniko Fixed with 1.2.2-SNAPSHOT

**Comment by domkun** [comment]
Works. Thank you.
